include/boost/geometry/geometry.hpp

```diff
@@ -20,11 +20,11 @@
 #ifndef BOOST_GEOMETRY_GEOMETRY_HPP
 #define BOOST_GEOMETRY_GEOMETRY_HPP
 
+#if !defined(BOOST_GEOMETRY_DISABLE_DEPRECATED_03_WARNING)
 #include <boost/config.hpp>
+#if defined(BOOST_NO_CXX14_CONSTEXPR)
 #include <boost/config/pragma_message.hpp>
-#if defined(BOOST_NO_CXX11_VARIADIC_TEMPLATES) || defined(BOOST_NO_CXX11_RVALUE_REFERENCES) || defined(BOOST_NO_CXX11_HDR_MEMORY)
-#if !defined(BOOST_GEOMETRY_DISABLE_DEPRECATED_03_WARNING)
-BOOST_PRAGMA_MESSAGE("CAUTION: Boost.Geometry in Boost 1.73 deprecates support for the C++03 and will require C++14 from Boost 1.75 onwards.")
+BOOST_PRAGMA_MESSAGE("CAUTION: Boost.Geometry in Boost 1.73 deprecates support for C++03 and will require C++14 from Boost 1.75 onwards.")
```

Comment:
It is possible to use Boost.Geometry without including `geometry.hpp` - we do that often ourselves in unit tests. And in the examples.
But in practice it is probably nearly always included like this.

So I'm OK with this place - but we might also place it in `boost/geometry/core/config.hpp` (including `boost/config.hpp` already)

Reply:
That's a good idea, though `core/config.hpp` is included only by a handful of files basically related to `get_turns` and tuples handling. So a different core file could be used, e.g. `core/tags.hpp` which at first glance (https://github.com/awulkiew/cxx-dependency-visualizer) is one of the most included bottom-level headers.

Reply:
I think we can keep it in `geometry.hpp`. There will be also docs, README, and other means to announce.